tsdb/chunks/chunks.go

```diff
@@ -220,19 +220,20 @@ func (w *Writer) cut() error {
 func cutSegmentFile(dirFile *os.File, magicNumber uint32, chunksFormat byte, allocSize int64) (headerSize int, newFile *os.File, seq int, err error) {
 	p, seq, err := nextSequenceFile(dirFile.Name())
 	if err != nil {
-		return 0, nil, 0, err
+		return 0, nil, 0, errors.Wrap(err, "next sequence file")
 	}
-	f, err := os.OpenFile(p, os.O_WRONLY|os.O_CREATE, 0666)
+	ptmp := p + ".tmp"
+	f, err := os.OpenFile(ptmp, os.O_WRONLY|os.O_CREATE, 0666)
 	if err != nil {
-		return 0, nil, 0, err
+		return 0, nil, 0, errors.Wrap(err, "open temp file")
```

Comment:
I think, after this block (i.e. after line 229), we should have a `defer` for closing and deleting the tmp file in case any of the errors below trigger, to not leak file descriptors and leave tmp files behind. (In case the closing or renaming has already happened, the close or delete call will return an error, but we don't mind then. You could also guard the deferred closing and deletion with a boolean to avoid spurious calls.)

Reply:
Good point, I have updated the PR